xarray/tests/test_plot.py

```diff
@@ -2381,3 +2382,50 @@ def test_facetgrid_single_contour():
     ds["time"] = [0, 1]
 
     ds.plot.contour(col="time", levels=[4], colors=["k"])
+
+
+@requires_matplotlib
+@pytest.mark.parametrize("figsize", [[4, 4], None])
+@pytest.mark.parametrize("aspect", [4 / 3, None])
+@pytest.mark.parametrize("size", [200, None])
+def test_get_axis(figsize, aspect, size):
+    # test get_axis works with different args combinations
+    # and return the right type
+
+    ax = "something"
+    if figsize is not None:
```

Comment:
I think this will be more readable if we didn't use `parametrize` so the tests look like

``` python
with pytest.raises(ValueError, match="both `figsize` and `ax`"):
        get_axis(figsize=[4,4], size=200, aspect=None, ax=ax)
```
and so on...

Reply:
that's 8 tests and would make the test pretty long so I think there's no perfect solution